control-plane/consul/resource_client_test.go

```diff
@@ -36,7 +36,15 @@ func Test_NewResourceServiceClient(t *testing.T) {
 
 	opts := hclog.LoggerOptions{Name: "resource-service-client"}
 	logger := hclog.New(&opts)
-	client, watcher, err := NewResourceServiceClient(context.Background(), discoverConfig, logger, serverConfig.Ports.GRPCTLS)
+
+	watcher, err := discovery.NewWatcher(context.Background(), discoverConfig, logger)
+	require.NoError(t, err)
+	require.NotNil(t, watcher)
+
+	defer watcher.Stop()
+	go watcher.Run()
+
+	client, err := NewResourceServiceClient(watcher)
 	require.NoError(t, err)
 	require.NotNil(t, client)
 	require.NotNil(t, watcher)
```

Comment:
~ Might be a diff issue on my end, but looks like maybe there's a duplicate `defer watcher.Stop()` under this block of code (outside comment range)?

Reply:
Correct! Fixed.